tests/numpy/test_gen_data.py

```diff
@@ -185,12 +185,14 @@ def test_can_turn_off_subarrays(dt):
         assert field.shape == ()
 
 
-@given(nps.integer_dtypes(endianness='>'))
-def test_can_restrict_endianness(dt):
-    if dt.itemsize == 1:
-        assert dt.byteorder == '|'
+@pytest.mark.parametrize('byteorder', ['<', '>'])
+@given(data=st.data())
+def test_can_restrict_endianness(data, byteorder):
+    dtype = data.draw(nps.integer_dtypes(byteorder, sizes=(16, 32, 64)))
+    if dtype.isnative:
+        assert dtype.byteorder == '='
```

Comment:
This assertion seems a bit odd to me, since it's really just checking numpy's claim that `=` means native, right?

Might be overkill, but using `sys.byteorder` to decide whether we expect `dt.byteorder` should be `=` or `byteorder` seems like it would make the intent really clear, and let you skip this bit.